cypress/e2e/page/download-pagetests.cy.js

```diff
@@ -21,29 +21,54 @@ describe("tests on download page ui via programmatic state loading and store int
 
             beforeEach(() => {
 
-            });
+                // 1. Open index page
+                cy.visit("/");
 
-            it("", () => {
+                // 2. Load test data
+                cy.loadTestDataIntoStore(p_dataset);
 
-            });
+                // 3. Move to the download page
+                cy.window().its("$nuxt.$router").then(router => {
 
-            it("", () => {
+                    // A. Route to annotation page
+                    router.push({ path: "/download" });
 
-            });
+                    // B. Once routing is complete, set the current page in the nuxt store
+                    // (normally would happen via navigation clicks)
+                    cy.commitToVuexStore("setCurrentPage", "download");
+                });
 
-            it("", () => {
+                // 4. Categories required for this test and the number of required columns for each category
+                const testCriteria = {
 
-            });
+                    categories: [
 
-            it("", () => {
+                        ["Subject ID", 1],
+                        ["Age", 1],
+                        ["Sex", 1]
+                    ]
+                };
 
+                // 5. Load the app with test criteria using the dataset
+                cy.loadAppState("download", p_dataset, testCriteria);
             });
 
-            it("", () => {
+            it("Clicking download button downloads annotated data dictionary JSON", () => {
+
+                // 0. Checking downloads folder state before click
+                cy.task("downloads", "cypress/downloads").then(folderStateBefore => {
 
+                    // 1. Download the annotated data dictionary in Neurobagel JSON format
+                    cy.get("[data-cy='download-button']")
+                        .click();
+
+                    // 2. Check number of files in downloads folder has increased by one
+                    cy.task("downloads", "cypress/downloads").then(folderStateAfter => {
+
+                        expect(folderStateAfter.length).to.be.eq(folderStateBefore.length + 1);
```

Comment:
If I understand correctly, the `folderStateAfter` array will have the filename in it. Might be worth also asserting over the filename as well, since this is an e2e test (maybe without the datetime addition)?

Reply:
It's worth it to spend a few minutes checking this out. The reason I'm not using Cypress' `readFile` here to check for a file in the downloads path is because our filenames are dynamic in that they contain a timestamp.

However we should be able to use `expect(x).to.contain(first_part_of_filename]`. The possible point of failure there for the test is if the same data dictionary is annotated and downloaded, it would contain the same first part of the filename sans timestamp.

Reply:
I have found a way to ensure the download folders is clear for tests such as this so the latter test failure possibility has been eliminated.

See: https://codenboxautomationlab.com/how-to-delete-previous-downloaded-files-in-cypress/

Reply:
After talking with @surchs, instead of the more risky file removal solution I have opted to check to see that the last file added to the `cypress/downloads` folder contains the proper file prefix as saved in `dataDictionary.filename`. Since there were no other tests using the `getVuexStoreValue` method, I have also altered it to look directly at `store.state` instead of trying to look for a getter (an old store paradigm we were using at the time of the method's creation.)